src/test/scala/scala/compat/java8/StepConvertersTest.scala

```diff
@@ -0,0 +1,572 @@
+package scala.compat.java8
+
+import org.junit.Test
+import org.junit.Assert._
+
+class StepConvertersTest {
+  import java.util._
+  import collectionImpl._
+  import converterImpl._
+  import StreamConverters._   // Includes StepConverters!
+  import scala.{ collection => co }
+  import collection.{ mutable => cm, immutable => ci, concurrent => cc }
+
+  def isAcc[X](x: X): Boolean = x match {
+    case _: AccumulatorStepper[_] => true
+    case _: DoubleAccumulatorStepper => true
+    case _: IntAccumulatorStepper => true
+    case _: LongAccumulatorStepper => true
+    case _ => false
+  }
+
+  def isLin[X](x: X): Boolean = x match {
+    case _: AbstractStepsLikeIterator[_, _, _] => true
+    case _: AbstractStepsWithTail[_, _, _] => true
+    case _ => false
+  }
+
+  trait SpecCheck { def apply[X](x: X): Boolean }
+  object SpecCheck {
+    def apply(f: Any => Boolean) = new SpecCheck { def apply[X](x: X): Boolean = f(x) }
+  }
+
+  def _eh_[X](x: => X)(implicit correctSpec: SpecCheck) { 
+    assert(x.isInstanceOf[Stepper[_]])
+    assert(correctSpec(x))
+  }
+
+  def IFFY[X](x: => X)(implicit correctSpec: SpecCheck) {
+    assert(x.isInstanceOf[Stepper[_]])
+    assert(correctSpec(x))
+    assert(isAcc(x))
+  }
+
+  def Okay[X](x: => X)(implicit correctSpec: SpecCheck) {
+    assert(x.isInstanceOf[Stepper[_]])
+    assert(correctSpec(x))
+    assert(!isAcc(x))
+    assert(isLin(x))
+  }
+
+  def Fine[X](x: => X)(implicit correctSpec: SpecCheck) {
+    assert(x.isInstanceOf[Stepper[_]])
+    assert(correctSpec(x))
+    assert(!isAcc(x))
+  }
+
+  def good[X](x: => X)(implicit correctSpec: SpecCheck) {
+    assert(x.isInstanceOf[Stepper[_]])
+    assert(correctSpec(x))
+    assert(!isAcc(x))
+    assert(!isLin(x))
+  }
+
+  def Tell[X](x: => X)(implicit correctSpec: SpecCheck) {
+    println(x.getClass.getName + " -> " + isAcc(x))
+    assert(x.isInstanceOf[Stepper[_]])
+    assert(correctSpec(x))
+  }
```

Comment:
Some reuse could make the seemingly arbitrary methods easier to understand, e.g.:

```
  def Okay[X](x: => X)(implicit correctSpec: SpecCheck) {
    Fine(x)
    assert(isLin(x))
  }

  def good[X](x: => X)(implicit correctSpec: SpecCheck) {
    Fine(x)
    assert(!isLin(x))
  }
```